### PR TITLE
Bump up min. PHP version to 8.1.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
Can't figure out why that fatal error is thrown for `MysqlTest`. I am unable to reproduced it locally.

Also don't understand why seemingly unrelated tests are throwing error for `memcached` loading on Windows.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -129,16 +127,16 @@ jobs:
...
+      if: matrix.php-version == '8.1'
+      uses: codecov/codecov-action@v2
 
   testsuite-windows:
     runs-on: windows-2019
-    name: Windows - PHP 8.0 & SQL Server
+    name: Windows - PHP 8.1 & SQL Server
 
     env:
-      EXTENSIONS: mbstring, intl, apcu, memcached, redis, pdo_sqlsrv
```
Comment (markstory) [inline .github/workflows/ci.yml:140]: I guess we'll have to wait for a new release to be made for this extension?

File: tests/TestCase/I18n/DateTimeTest.php [hunk 2]
```diff
@@ -507,7 +510,11 @@ public function testToString(): void
         $time = new DateTime('2014-04-20 22:10');
         DateTime::setDefaultLocale('fr-FR');
         DateTime::setToStringFormat(IntlDateFormatter::FULL);
-        $this->assertTimeFormat('dimanche 20 avril 2014 22:10:00 UTC', (string)$time);
+        $expected = 'dimanche 20 avril 2014 22:10:00 UTC';
+        if (PHP_OS_FAMILY === 'Windows') {
+            $expected = 'dimanche 20 avril 2014 22:10:00 temps universel coordonné';
```
Comment (markstory) [inline tests/TestCase/I18n/DateTimeTest.php:515]: I wonder if all windows installs will have this value? :thinking: We could switch to a regex in the future if need be.

**Review by markstory — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ADmad) [reply .github/workflows/ci.yml:140]: Yup we need to wait for a stable release of memcached with PHP 8.1 support for Windows.

File: tests/TestCase/I18n/DateTimeTest.php (hunk 2 above)
Comment (ADmad) [reply tests/TestCase/I18n/DateTimeTest.php:515]: Yeah this is a weird one, we can adjust as needed in future.

**Pull request merged**